event-dispatcher/src/main/java/io/sk8s/event/dispatcher/KafkaConsumerMonitor.java

```diff
@@ -0,0 +1,193 @@
+	}
+
+	public synchronized void beginTracking(String group, String topic) {
+		if (logEndOffsetTrackingConsumer == null) {
+			logEndOffsetTrackingConsumer = createNewConsumer(monitoringConsumerGroupID);
+		}
+		consumersByGroup.computeIfAbsent(group, k -> createNewConsumer(group));
+		tracked.add(new TopicGroup(topic, group));
+		reassignPartitions();
+	}
+
+	public synchronized void stopTracking(String group, String topic) {
+		tracked.remove(new TopicGroup(topic, group));
+		if (tracked.stream().noneMatch(tg -> tg.group.equals(group))) {
+			consumersByGroup.remove(group).close();
+		}
+		reassignPartitions();
+	}
+
+	private void reassignPartitions() {
+		logEndOffsetTrackingConsumer.assign(
+				tracked.stream()
+						.flatMap(tg -> logEndOffsetTrackingConsumer.partitionsFor(tg.topic).stream()
+								.map(pi -> new TopicPartition(tg.topic, pi.partition()))
+						).collect(Collectors.toSet()));
+	}
+
+	public Map<TopicGroup, List<Offsets>> compute() {
+		logEndOffsetTrackingConsumer.seekToEnd(FOR_ALL_PARTITIONS);
+
+		return tracked.stream()
+				.collect(Collectors.toMap(
+						Function.identity(),
+						tg -> {
+							List<PartitionInfo> partitionInfos = logEndOffsetTrackingConsumer.partitionsFor(tg.topic);
+							return partitionInfos.stream()
+									.map(pi -> new TopicPartition(tg.topic, pi.partition()))
+									.map(
+											tp -> {
+												OffsetAndMetadata committed = consumersByGroup.get(tg.group).committed(tp);
+												if (committed == null) {
+													committed = new OffsetAndMetadata(0L);
+												}
+												long position = logEndOffsetTrackingConsumer.position(tp);
+												return new Offsets(tp.partition(), position,
+														committed.offset());
+											})
+									.collect(Collectors.toList());
+						}));
+	}
+
+	static class TopicGroup {
+		final String topic;
+	
+		final String group;
+
+		TopicGroup(String topic, String group) {
+			this.topic = topic;
+			this.group = group;
+		}
+
+		@Override
+		public boolean equals(Object o) {
+			if (this == o) return true;
+			if (o == null || getClass() != o.getClass()) return false;
+			TopicGroup that = (TopicGroup) o;
+			return Objects.equals(topic, that.topic) &&
+					Objects.equals(group, that.group);
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(topic, group);
+		}
+
+		@Override
+		public String toString() {
+			return "TopicGroup{" +
+					"topic='" + topic + '\'' +
+					", group='" + group + '\'' +
+					'}';
+		}
+	}
+
+	static class Offsets {
+		final int partition;
+		final long endOffset;
+		final long currentOffset;
+
+		Offsets(int partition, long endOffset, long currentOffset) {
+			this.partition = partition;
+			this.endOffset = endOffset;
+			this.currentOffset = currentOffset;
+		}
+
+		long getLag() {
+			return endOffset - currentOffset;
+		}
+
+		@Override
+		public String toString() {
+			return "Offsets{" +
+					"partition=" + partition +
+					", endOffset=" + endOffset +
```

Comment:
the "end" is not actually an offset

Reply:
actually I now see where they use the term "log-end-offset"... think I see where you're coming from now (also on the name of the consumer)